### PR TITLE
pd/client: add more retries

### PR DESCRIPTION
**Pull request by huachaohuang**
We may fail if we start PD and TiKV together because TiKV cannot get cluster ID from PD.
We can retry for a long time when creating the client.

### CODE REVIEW

File: src/pd/client.rs [hunk 1]
```diff
@@ -153,8 +152,20 @@ impl RpcClient {
             core: Mutex::new(RpcClientCore::new(endpoints)),
             cluster_id: 0,
         };
-        client.cluster_id = try!(client.get_cluster_id());
-        Ok(client)
+
+        for _ in 0..MAX_PD_SEND_RETRY_COUNT {
```
Comment (siddontang) [inline src/pd/client.rs]: can we use a back-off algorithm instead. I think 1s is long here.

Comment (huachaohuang) [inline src/pd/client.rs]: I think 1s is fine, we only retry here when TiKV starts and cannot connect to PD. PD may not ready to serve yet, no need to retry in a very short time.

Comment (siddontang) [inline src/pd/client.rs]: Got it.

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @overvenus

**Pull request merged**